### PR TITLE
Grenade angles

### PR DESCRIPTION
**Pull request by hettoo**
To help players fight gravity when shooting grenades, qfusion offsets the angle at which the grenade is shot a little upwards. Currently, this offset is a constant of 5 degrees, which means that it is impossible to shoot straight down.

Thinking about the intended effect of compensating gravity, it seems more natural to regulate the offset according to the extent to which the viewing direction is orthogonal to the direction of gravity.
Thus, I propose to turn the adjustment into a monotone bijection by means of the cosine function.

Initially, the idea was to use this for grenade jumps as it enables shooting straight down, but the grenade settings are not so well suited to make a nice jump. Still, I find the change satisfying conceptually.

Of course, the intuition of compensating gravity is a bit off considering that the trajectory is often much longer when shooting somewhat up than when shooting somewhat down, but that cannot really (and should not) be accounted for.

Note that due to the cosine the effect does not diminish very quickly around the horizon (25 degrees up/down gives an offset of 4.5 instead of 5), so I expect players will not really notice the change. The value 5 could be raised a little to make it on average more like the old situation.

(The thing is bijective on [-90, 90] as long as that value 5 is not raised above 57 something.)

### CODE REVIEW

File: source/game/g_weapon.cpp [hunk 1]
```diff
@@ -725,16 +725,7 @@ edict_t *W_Fire_Grenade( edict_t *self, vec3_t start, vec3_t angles, int speed,
 		damage = 9999;
 
 	if( aim_up )
-	{
-		angles[PITCH] -= 5; // aim some degrees upwards from view dir
-
-		// clamp to front side of the player
```
Comment (Triang3l) [inline source/game/g_weapon.cpp:731]: Was removing this necessary?

Comment (hettoo) [inline source/game/g_weapon.cpp:731]: Not necessary, but harmless too.
PM_ApplyMouseAnglesClamp ensures that the angle is in [-90, 90], and as I said unless the value 5 is raised to a huge offset of >57 the result after the new modification is also in that range.

**Comment by viciious** [comment]
So, in the end, how does this patch affect grenades movement and how that helps race?

**Comment by hettoo** [comment]
The angle at which the grenade is thrown is tampered with less the more you look up or down from the horizon. The main advantage of this is that players can shoot straight down, e.g. to do a grenade jump or scare off an opponent. Originally, the fixed 5 degree offset would enter in and for the jump you would have to run after your grenade.

Another way to achieve the same effect is to get rid of the angle tampering completely and instead offset the position at which grenades are thrown a little upwards. However, the effect in this case is much weaker, depending on the distance that the grenade is to travel, so I would prefer the present solution that is closer to the original situation.

**Pull request merged**